### ISSUE TITLE
Dev docs show previous stable release notes

### ISSUE THREAD

**Issue by moi90** [scikit-image/scikit-image#6496]
## Description

While working on #6495, I noticed that when I view https://scikit-image.org/docs/dev/release_notes.html, it says "Docs for 0.20.0.dev0" in the side bar on the left, but the text in `release_notes.html` is for the previous stable release v0.19.3.

Is this intentional?

I was half-expecting the dev docs to display the current work-in-progress of the dev "release notes". But I also acknowledge that this would be somewhat strange...

**Comment by mkcor** [comment]
Dear @moi90,

You're making a good point. I also think it would be strange to display 'work-in-progress' 'release notes' ('work in progress' and 'release' being contradictory). I would suggest renaming the "[Release notes](https://scikit-image.org/docs/dev/release_notes.html#release-notes)" header into "Latest release notes."

**Comment by mkcor** [comment]
@moi90 would it be fine to close this issue?

**Comment by moi90** [comment]
Hmm, on https://scikit-image.org/docs/dev/ it still says: "View the [release notes](https://scikit-image.org/docs/dev/release_notes.html) for 0.21.0rc0.dev0." But I think for a dev version, one can expect a little bit roughness...

**Comment by moi90** [comment]
But still, I would rather expect the work-in-progress release notes for the upcoming release, e.g. "Announcement: scikit-image 0.21.0 (unreleased)" (doc/release/release_dev.rst). If I want to see the latest stable release notes, I visit the documentation of the previous release, not the dev version.

**Comment by mkcor** [comment]
@moi90 ok, I guess pointing to [release_dev.rst](https://github.com/scikit-image/scikit-image/blob/main/doc/release/release_dev.rst) instead of the latest release notes should be feasible... Are you able to submit a PR?

**Comment by moi90** [comment]
Unfortunately no, not in the foreseeable future...

**Comment by rum1887** [comment]
> > Unfortunately no, not in the foreseeable future...
> 
> No worries, I completely understand!
> 
> @rum1887 is there a chance you could help out with this? 😉

sure!

**Comment by lagru** [comment]
For reference, it's https://github.com/scikit-image/skimage-web/issues/83.

**Comment by lagru** [comment]
See also #6714, which proposes to resolve this currently by [showing all release notes and marking the unreleased notes](https://output.circle-artifacts.com/output/job/a2fe11de-c159-48d2-bee9-44105163c89c/artifacts/0/doc/build/html/release_notes/index.html).

**Comment by lagru** [comment]
@rum1887, you say

> I have moved the issue to scikit-image/ scikit-web

Do you mean https://github.com/scikit-image/skimage-web/issues/83? I think that one is unrelated to this issue.

**Comment by rum1887** [comment]
> @rum1887, you say
> 
> > I have moved the issue to scikit-image/ scikit-web
> 
> Do you mean https://github.com/scikit-image/skimage-web/issues/83? I think that one is unrelated to this issue.

I'm sorry for the confusion, I'll look into this

**Comment by lagru** [comment]
Yes, I think most of the critique was addressed with the move to the new PyData theme.

Maybe something similar to the format that @jarrodmillman is using here
https://github.com/scikit-image/scikit-image/blob/4e4309edc53afc713b7f01e58b6d6b518391f57e/skimage/__init__.py#L68

I can't find the reference right now, but I think sphinx supports something like `|version|` to reference the current version from `conf.py`. That should probably be used in the title for the release notes document.
